### ISSUE TITLE
persistent WSL image size after importing

### ISSUE THREAD

**Issue by joelguerrero** [microsoft/WSL#8672]
Hi,

I resized a WSL image to 512M, then I exported the image, and imported in another computer. 

But when I open the new imported image, the size is back to 256M.  I wonder if it is possible to have a persistent size when importing the image? or Am I missing something?

Thanks for your help,

JG

**Comment by NotTheDr01ds** [comment]
There's no way that I'm aware of to change the default `vhdx` size on WSL, but note that [WSL release 0.58.0](https://github.com/microsoft/WSL/releases/tag/0.58.0) does increase the default to 1TB.

You don't mention your Windows release (please remember to use the issue "template" when reporting), but Windows 11 users can install the latest release through the Microsoft Store using the "Windows Subsystem for Linux Preview."

**Comment by joelguerrero** [comment]
Hi,

Sorry for not using the template.  I had my head in something else and I didn't pay attention to that.

I am using windows 11, and opensuse 15.4 as the linux os.

Increasing the size of the virtual disk is relatively easy, I just followed these steps:
https://docs.microsoft.com/en-us/windows/wsl/vhd-size

My issue is that the size is not persistent. If I import the image in another computer, I always get 250GB and not the size of the exported image (500GB).  Again, this is not an issue because increasing the size is easy. 

What will happen if I try to export an image with a size larger than 250GB? 

How do I install this preview version? I didnt find WSL preview in the store.

Currently I am using this kernel: 5.10.102.1

**Comment by NotTheDr01ds** [comment]
> What will happen if I try to export an image with a size larger than 250GB?

Hmm - good question.  I haven't tried it, but I would expect the image would *export* just fine, since the export operation is essentially just a bsdtar of the relevant contents.  However, I believe that the resulting `--import` would be problematic -- it would probably run out of space on the GA release with the 250GB limit.  I don't expect that there's any mechanism in place to expand the size of the VHDX when `--import`'ing, but I could be wrong.

> How do I install this preview version? I didnt find WSL preview in the store.

It should be visible to Windows 11 users, but hidden for Windows 10 users.  [Here's](https://aka.ms/wslstorepage) a direct link that one of the WSL team members posted in another issue here.

I also found a way to install it on Windows 11 without Store access, which I go into (with warnings) [in this issue](https://github.com/microsoft/WSL/issues/7506#issuecomment-1198513821).

**Comment by Biswa96** [comment]
> I also found a way to install it on Windows 11 without Store access, which I go into with warnings

AFAIK, any Windows/Microsoft Store package can be installed without Store access and it existed since inception of Store. There is **no** warning. The package in GitHub release page is provided by Microsoft developers, so you are good to go.

**Comment by NotTheDr01ds** [comment]
@Biswa96 I agree, it's probably fine.  The only thing that made me think these may be slightly different than "normal" store packages is that, when attempting to double-click one to launch in the App Installer, it gives the message:

> `This app package is not supported for installation by App Installer because it uses certain restricted capabilities.`

It's still installable through the CLI in an admin shell, so it's possible that the App Installer just doesn't elevate UAC.

**Comment by Biswa96** [comment]
> Regardless, the other warning is that it probably shouldn't be installed on Windows 10.

Yes, there are some HCS (host compute system) and HCN (host compute network) APIs and JSON fields used by lifted WSL package which are not available in Windows 10.

**Comment by joelguerrero** [comment]
After rebooting the computer, now I see the preview version. I don't know why I wasn't able to see it before.

I can confirm that the size of the virtual HD in the preview version is 1TB by default. I guess the steps to expand the size are the same. 

Out of curiosity, what if I want to reduce the size of the virtual HD? same steps as for expanding the virtual HD?

Is there any documentation for the WSL preview version?

JG

**Comment by NotTheDr01ds** [comment]
> Is there any documentation for the WSL preview version?

Documentation is mostly in the form of [release notes](https://github.com/microsoft/WSL/releases).

**Comment by NotTheDr01ds** [comment]
> what if I want to reduce the size of the virtual HD?

Reclaiming unused space from the sparse VHD can be done with the `Optimize-VHD` cmdlet.  See #4699 and [this answer](https://superuser.com/a/1612289/1210833) I gave on Super User.

If you wanted to reduce the maximum size of the drive, then yes, I'm assuming that you could do that with `Resize-VHD`, given that the contents didn't exceed the new size.  However, I haven't tried it with an ext4 filesystem, so I'm not 100% sure.
